### ISSUE TITLE
Test failures with git 2.29

### ISSUE THREAD

**Issue by simark** [AdaCore/git-hooks#16]
On my home computer that has git 2.29.2 (latest as of today), I see more failures than on Ubuntu 20.04, which has git 2.25.4:

```
Testsuite Results Summary -- Out of 203 testscase(s) run in total:

      10 testcase(s) failed
       0 testcase(s) crashed
```

One such failure is:

```
./run-testsuite --diffs -v tests/T314-004__new_branch_with_unusual_ref_name

--- expected
+++ output
@@ -56,2 +56,2 @@
 To ../bare/repo.git
- * [new branch]      master -> refs/for/master
+ * [new reference]   master -> refs/for/master
```

So, it's perhaps just harmless terminology changes in the newer git version.

**Comment by simark** [comment]
The good news is that I've manually built upstream git 2.25.4 on my home computer, and I get 0 failures with it!

**Comment by brobecke** [comment]
Good news indeed, @simark .

You've discovered a known weakness of the testsuite: It more or less assumes a specific version of Git, because it matches Git's output almost verbatim. I say almost, because I tried to add some post processing to make it work with multiple versions of Git, but in the end, there were some differences which I just couldn't figure out how to automate. The one above, I think we might be able to automate it, not sure. But the main point is that I know about this weakness, and tried a few times to think about how I could avoid it without having to introduce too much complication. In the end, I couldn't really thing of something, and since I was the only one running the testsuite anyway, I ended up doing nothing about it.

**Comment by brobecke** [comment]
Hi @simark . Just a quick update to let you know that this should now be fixed.

As I mentioned in my previous answer, the testsuite is not very flexible with the version of Git we can run it with, but for differences of output like that, we can try to add automatic adjustments that allow us to support both new and past versions... I've created a basic system that allows us to massage the output expected with recent versions to match the output expected by less recent ones. We'll see how many fixups we'll need in practice and whether that works well enough.

**Comment by brobecke** [comment]
I forgot to mention which commit fixed the issue: Commit 09381ab4a10306eda5d3cccdd05d185716fc7d80.
